Add "making test data files" section to contributing guide (#233)
hyperspy/rosettasciio: CONTRIBUTING.rst

- give a minimal example demonstrating the bug,
- copy and paste the error traceback.

.. _making_test_files:

Making test data files
======================

> Test data fales are typically generated using third party software, for example using a proprietary

Reviewer: ```suggestion
Test data files are typically generated using third party software, for example using a proprietary
```

software on a scientific instrument. These files are added to the `test suite <https://en.wikipedia.org/wiki/Test_suite>`_

> of RosettaSciIO to make sure that future code development will not introduce bug or feature 
> regression. It is important that the test data files area as small as possible to avoid to work

Reviewer: ```suggestion
of RosettaSciIO to make sure that future code development will not introduce bugs or feature 
regressions. It is important that the test data files area as small as possible to avoid to work
```

> regression. It is important that the test data files area as small as possible to avoid to work
> with GBs large repository of test data. Indeed, the test suite is made of severals hundreds of

Reviewer: ```suggestion
regression. It is important that the test data files area as small as possible to avoid working
with a repository that contains GBs of test data. Indeed, the test suite is made of severals hundreds of
```

> with GBs large repository of test data. Indeed, the test suite is made of severals hundreds of

Reviewer: ```suggestion
with a repository that contains GBs of test data. Indeed, the test suite is made of severals hundreds of
```

test data files and this number of files will keep growing as new features and formats are added
to RosettaSciIO.

> User can contribute by generating these files on softwares they have access to and by making these

Reviewer: ```suggestion
Users can contribute by generating these files on softwares they have access to and by making these
```

files available openly; then a RosettaSciIO developer will help with adding these data to the test suite.

> What makes good test data files:

Reviewer: ```suggestion
What characterizes good test data files:
```

> - Relevant features: the test data files doesn't need to contains any meaningfull data but they need to

Reviewer: ```suggestion
- Relevant features: the test data files do not need to contain any meaningful data, but they need to
```

  cover as much as possible of the format functionalities.

> - Small in size:

Reviewer: ```suggestion
- Small size:
```

>   - Acquire minimum number of pixels or channels; in case of images or spectrum images acquire non-square

Reviewer: ```suggestion
  - Acquire minimum number of pixels or channels. In case of maps or spectrum images acquire a non-square grid
```

>     (e.g. "x" and "y" have different length).

Reviewer: ```suggestion
    (e.g. "x" and "y" have different lengths).
```

>   - Generate containing no signal (e.g. zeros) as files containing only very few values will compress very well.

Reviewer: ```suggestion
  - If possible, generate data that contains no signal (e.g. zeros) as files containing only very few values will compress very well.
```
